samples/cellular/nrf_cloud_multi_service/src/application.c

```diff
@@ -39,6 +41,14 @@ BUILD_ASSERT(CONFIG_AT_CMD_REQUEST_RESPONSE_BUFFER_LENGTH >= AT_CMD_REQUEST_ERR_
 #define TEMP_ALERT_HYSTERESIS 1.5f
 #define TEMP_ALERT_LOWER_LIMIT (TEMP_ALERT_LIMIT - TEMP_ALERT_HYSTERESIS)
 
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+#define MSG_OBJ_DEFINE(_obj_name) \
+	NRF_CLOUD_OBJ_JSON_DEFINE(_obj_name);
+#elif defined(CONFIG_NRF_CLOUD_COAP)
+#define MSG_OBJ_DEFINE(_obj_name) \
+	NRF_CLOUD_OBJ_COAP_CBOR_DEFINE(_obj_name);
+#endif
```

Comment:
I blame Justin ;-)

Reply:
Not sure why this would go in nrf_cloud...
But put it wherever.

Reply:
> Not sure why this would go in nrf_cloud...

It's a generic object declarator that intelligently chooses between either the COAP or JSON object type, and to me that choice feels like the domain of nrf_cloud rather than of the application

I don't see a reason to **_force_** that choice to be made by nrf_cloud always, but it does seem like it could live in `nrf_cloud` as `NRF_CLOUD_OBJ_AUTO_DEFINE` 

This also makes me wonder: Should `nrf_cloud_data_type` instead be `nrf_cloud_obj_fixed_data_type` or something to that effect?

Reply:
this is a demo application which now supports 2 build options.
i don't think many real world applications will be built that way.
having an "auto" macro seems like it would just lead to confusion and/or lack of awareness of what is being done.
but, again, put it wherever you want.  this is not worth a continued discussion on my part.

Reply:
Sure, `application.h` is fine if you both feel that way, but maybe let's reconsider it down the line when COAP is more mature